src/Map.php

```diff
@@ -58,6 +58,19 @@ public function add(MapInterface $map) {
     }
 
     /**
+     * SetState 
+     *
+     * Support reloading class via var_export definition.
+     * 
+     * @param array $mapData Array of logger data needed to reconsturct logger
+     * @return string        Implementaiton of logger class to be passed to the Map class
+     */
+    public static function __set_state($mapData) {
+        $map = new Map($mapData['map'], $mapData['logger']);
```

Comment:
@mglinski This should be changed to:

```php
public static function __set_state(array $mapData) {
    return new static($mapData['map'], $mapData['logger']);
}
```

Otherwise extended classed will be converted to the parent class: http://codepad.viper-7.com/9dmDCy